10up/distributor
# Remove `-dev` from our version number

### Description of the Change

In #210, we added `-dev` to our version number in order to output a warning for users that may be running this plugin straight from a non-built branch. Later we added more code around this to remove that `-dev` during release. There's nothing wrong with this approach but does cause some confusion during the release process and having to remember when to add that `-dev` or when to remove it.

This PR changes that up to instead look for the existence of the `composer.lock` file. If that exists, we then output the same warning message. This file should only exist in branches that aren't built, for instance `stable` won't contain that file. Since we're not relying on that `-dev` string anymore, I also removed any code that was in place to support that.

Closes #820

### Alternate Designs

Could leave as-is and just ensure we better document why we have this

### Possible Drawbacks

None

### Verification Process

1. Check this branch out
2. Ensure you still see a warning message when visiting any Distributor specific pages (like the settings page or Pull screen)
3. For more testing, run `npm run release`, which will build a `release` directory
4. Move this directory to the root of the `plugins` directory, temporarily replacing the  current `distributor` directory
5. You should now be running the "released" version of Distributor. Ensure you don't see that warning anymore and also don't have any other issues

### Checklist:


### Changelog Entry

> Changed - Update how we check if someone is running a development version of Distributor
